### ISSUE TITLE
`cudagrad.seed` match PyTorch manual_seed for easier testing

### ISSUE THREAD

**Issue by yrmo** [yrmo/cudagrad#5]
https://pytorch.org/docs/stable/_modules/torch/random.html#manual_seed

```py
[[docs]](https://pytorch.org/docs/stable/generated/torch.manual_seed.html#torch.manual_seed)def manual_seed(seed) -> torch._C.Generator:
    r"""Sets the seed for generating random numbers. Returns a
    `torch.Generator` object.

    Args:
        seed (int): The desired seed. Value must be within the inclusive range
            `[-0x8000_0000_0000_0000, 0xffff_ffff_ffff_ffff]`. Otherwise, a RuntimeError
            is raised. Negative inputs are remapped to positive values with the formula
            `0xffff_ffff_ffff_ffff + seed`.
    """
    seed = int(seed)
    import torch.cuda

    if not torch.cuda._is_in_bad_fork():
        torch.cuda.manual_seed_all(seed)

    import torch.mps
    if not torch.mps._is_in_bad_fork():
        torch.mps.manual_seed(seed)

    _seed_custom_device(seed)

    return default_generator.manual_seed(seed)
```

Okay to avoid the licensing issue figure out what this calls into in... C++? CUDA has to know about this too, this might get real tricky...
